presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsMetadata.java

```diff
@@ -0,0 +1,375 @@
+import com.teradata.tpcds.column.StoreColumn;
+import com.teradata.tpcds.column.StoreReturnsColumn;
+import com.teradata.tpcds.column.StoreSalesColumn;
+import com.teradata.tpcds.column.TimeDimColumn;
+import com.teradata.tpcds.column.WarehouseColumn;
+import com.teradata.tpcds.column.WebPageColumn;
+import com.teradata.tpcds.column.WebReturnsColumn;
+import com.teradata.tpcds.column.WebSalesColumn;
+import com.teradata.tpcds.column.WebSiteColumn;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.spi.type.DecimalType.createDecimalType;
+import static com.facebook.presto.spi.type.VarcharType.createVarcharType;
+import static com.facebook.presto.tpcds.Types.checkType;
+import static java.util.Objects.requireNonNull;
+
+public class TpcdsMetadata
+        implements ConnectorMetadata
+{
+    public static final String TINY_SCHEMA_NAME = "tiny";
+    public static final double TINY_SCALE_FACTOR = 0.01;
+
+    public static final List<String> SCHEMA_NAMES = ImmutableList.of(
+            TINY_SCHEMA_NAME, "sf1", "sf10", "sf100", "sf300", "sf1000", "sf3000", "sf10000", "sf30000", "sf100000");
+
+    private final Set<String> tableNames;
+
+    public TpcdsMetadata()
+    {
+        ImmutableSet.Builder<String> tableNames = ImmutableSet.builder();
+        for (Table table : Table.getBaseTables()) {
+            tableNames.add(table.name().toLowerCase());
+        }
+        this.tableNames = tableNames.build();
+    }
+
+    @Override
+    public List<String> listSchemaNames(ConnectorSession session)
+    {
+        return SCHEMA_NAMES;
+    }
+
+    @Override
+    public ConnectorTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
+    {
+        requireNonNull(tableName, "tableName is null");
+        if (!tableNames.contains(tableName.getTableName())) {
+            return null;
+        }
+
+        // parse the scale factor
+        double scaleFactor = schemaNameToScaleFactor(tableName.getSchemaName());
+        if (scaleFactor < 0) {
+            return null;
+        }
+
+        return new TpcdsTableHandle(tableName.getTableName(), scaleFactor);
+    }
+
+    @Override
+    public List<ConnectorTableLayoutResult> getTableLayouts(
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            Constraint<ColumnHandle> constraint,
+            Optional<Set<ColumnHandle>> desiredColumns)
+    {
+        TpcdsTableHandle tableHandle = checkType(table, TpcdsTableHandle.class, "table");
+        String tableName = tableHandle.getTableName();
+        Map<String, ColumnHandle> columns = getColumnHandles(session, tableHandle);
+        Optional<ColumnHandle> partitioningColumn = Optional.empty();
+
+        if (tableName.equals(Table.STORE_SALES.getName())) {
+            partitioningColumn = Optional.of(columns.get(StoreSalesColumn.SS_TICKET_NUMBER.getName()));
```

Comment:
Yes, that's a good suggestion. However, since we're selecting the partitioning column based on the table name instead of the `Table` object we will still have this `if, else` block. I agree that this block is not very desirable and I'll try to come up with something better. @maciejgrzybek's static initialized map is a good idea.

I'll wait until the connector is done end to end to improve this section because I want to make sure the information I provide here will work with the execution part. I'll add a TODO.